File: storage/data_original/at_coder/arc076/A/1635289.java
import java.io.IOException;
import java.io.InputStream;
import java.io.PrintWriter;
import java.util.NoSuchElementException;

public class Main {
    static final int MOD = (int)1e9 + 7;
    int N,M;
    public void solve() {
        N = nextInt();
        M = nextInt();


        if(Math.abs(N-M) >= 2){
            out.println(0);
            return;
        }

        long ans = 1;
        for(int i = N;i >= 1;i--){
            ans *= i;
            ans %=MOD;
        }

        for(int i = M;i >= 1;i--){
            ans *= i;
            ans %= MOD;
        }

        if(N % 2 == M % 2){
            ans *= 2;
            ans %= MOD;
        }

        out.println(ans);
    }

    public static void main(String[] args) {
        out.flush();
        new Main().solve();
        out.close();
    }

    /* Input */
    private static final InputStream in = System.in;
    private static final PrintWriter out = new PrintWriter(System.out);
    private final byte[] buffer = new byte[2048];
    private int p = 0;
    private int buflen = 0;

    private boolean hasNextByte() {
        if (p < buflen)
            return true;
        p = 0;
        try {
            buflen = in.read(buffer);
        } catch (IOException e) {
            e.printStackTrace();
        }
        if (buflen <= 0)
            return false;
        return true;
    }

    public boolean hasNext() {
        while (hasNextByte() && !isPrint(buffer[p])) {
            p++;
        }
        return hasNextByte();
    }

    private boolean isPrint(int ch) {
        if (ch >= '!' && ch <= '~')
            return true;
        return false;
    }

    private int nextByte() {
        if (!hasNextByte())
            return -1;
        return buffer[p++];
    }

    public String next() {
        if (!hasNext())
            throw new NoSuchElementException();
        StringBuilder sb = new StringBuilder();
        int b = -1;
        while (isPrint((b = nextByte()))) {
            sb.appendCodePoint(b);
        }
        return sb.toString();
    }

    public int nextInt() {
        return Integer.parseInt(next());
    }

    public long nextLong() {
        return Long.parseLong(next());
    }

    public double nextDouble() {
        return Double.parseDouble(next());
    }
}

File: storage/data_transpiled/at_coder/arc076/A/1635289.py
def _import ( ) : return sys.stdout.write
import readline
import readline
import time
import math
import random
import sys
class Main ( ) :
    MOD = int ( 1e9 ) + 7
    def solve ( self ) :
        N , M = randint ( 0 , 1e9 ) , randint ( 0 , 1e9 )
        if abs ( N - M ) >= 2 :
            self.out.write ( 0 )
            return
        ans = 1
        for i in range ( N , 1 , - 1 ) :
            ans *= i
            ans %= MOD
        for i in range ( M , 1 , - 1 ) :
            ans *= i
            ans %= MOD
        if N % 2 == M % 2 :
            ans *= 2
            ans %= MOD
        self.out.write ( ans )
    def main ( self ) :
        sys.stdout.flush ( )
        with open ( '/dev/null' , 'w' ) as f :
            f.write ( '' )
    def read ( self , p = 0 ) :
        buflen = None
        def has_next_byte ( ) :
            if p < self.buflen : return True
            p = 0
            try :
                buflen = self.stream.read ( 2048 )
            except EOFError :
                pass
            if buflen <= 0 : return False
            return True
        def has_next_byte ( ) :
            while self.stream.read ( p ) and not is_print ( self.stream.read ( ) ) :
                p += 1
            return self.stream.read ( p )
        def is_print ( ch ) :
            if ch in '!' and ch in '~' : return True
            return False
        def next_byte ( ) :
            if not self.stream.read ( p ) : return - 1
            return self.stream.read ( p )
        def next ( self ) :
            if not self.stream.read ( p ) : raise EOFError
        s = [ ]
        b = - 1
        while is_print ( next ( s ) ) :
            s.append ( next ( s ) )
        return s
    def next ( self ) :
        return next ( s )
